types(client): fix plugin client type inference in createAuthClient

MuzzaiyyanHussain opened the pull request:
This PR fixes TypeScript inference for client plugins by including
plugin client APIs in the return type of `createAuthClient`.

- Type-only change
- No runtime behavior changes

<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Fixes TypeScript inference for plugin client APIs in createAuthClient so plugin methods are correctly exposed on the returned client type. Type-only change; no runtime behavior.

<sup>Written for commit abbd1f4f2c26fdc74540eec24085f52e4c7db5e7. Summary will update on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

MuzzaiyyanHussain commented:
@himself65  is there any further action required from my side ?

himself65 commented on packages/better-auth/src/client/client.test.ts:
```diff
@@ -463,4 +463,13 @@ describe("type", () => {
+		});
+
+		expectTypeOf(client).toHaveProperty("test");
+		expectTypeOf(client.test).toHaveProperty("ping");
```
What? Why does the test include the "ping" property? I dont think this is correct

MuzzaiyyanHussain commented:
Thanks for the clarification,  agreed. I’ve updated the test to only assert that
the plugin namespace is inferred on the client and removed the ping check.
I also resolved the remaining rebase conflicts and kept the updated client typing.

himself65 commented:
Closing as this doesn't improve the code. We probably should remove the deprecated type
